ui/src/app/routing/app-routing.tsx

```diff
@@ -0,0 +1,418 @@
+import {bindApiClients as notebooksBindApiClients} from 'app/services/notebooks-swagger-fetch-clients';
+import * as fp from 'lodash/fp';
+import outdatedBrowserRework from 'outdated-browser-rework';
+import {useEffect, useState} from 'react';
+import * as React from 'react';
+import {Switch, useHistory} from 'react-router-dom';
+import {StackdriverErrorReporter} from 'stackdriver-errors-js';
+
+import {
+  AppRoute,
+  AppRouter,
+  withRouteData
+} from 'app/components/app-router';
+import {NotificationModal} from 'app/components/modals';
+import {withRoutingSpinner} from 'app/components/with-routing-spinner';
+import {CookiePolicy} from 'app/pages/cookie-policy';
+import {SignIn} from 'app/pages/login/sign-in';
+import {NotFound} from 'app/pages/not-found';
+import {SessionExpired} from 'app/pages/session-expired';
+import {SignInAgain} from 'app/pages/sign-in-again';
+import {SignedIn} from 'app/pages/signed-in/signed-in';
+import {UserDisabled} from 'app/pages/user-disabled';
+import {disabledGuard, signInGuard} from 'app/routing/guards';
+import {bindApiClients, configApi, getApiBaseUrl, workspacesApi} from 'app/services/swagger-fetch-clients';
+import {useIsUserDisabled} from 'app/utils/access-utils';
+import {initializeAnalytics} from 'app/utils/analytics';
+import {useAuthentication} from 'app/utils/authentication';
+import {
+  cookiesEnabled,
+  LOCAL_STORAGE_API_OVERRIDE_KEY,
+  LOCAL_STORAGE_KEY_TEST_ACCESS_TOKEN
+} from 'app/utils/cookies';
+import {ExceededActionCountError, LeoRuntimeInitializer} from 'app/utils/leo-runtime-initializer';
+import {
+  currentWorkspaceStore,
+  nextWorkspaceWarmupStore, startTitleSetter,
+  urlParamsStore
+} from 'app/utils/navigation';
+import {
+  authStore,
+  runtimeStore,
+  serverConfigStore,
+  stackdriverErrorReporterStore, useStore
+} from 'app/utils/stores';
+import {environment} from 'environments/environment';
+import {Configuration} from 'generated/fetch';
+import 'rxjs/Rx';
+
+declare const gapi: any;
+
+const CookiePolicyPage = fp.flow(withRouteData, withRoutingSpinner)(CookiePolicy);
+const NotFoundPage = fp.flow(withRouteData, withRoutingSpinner)(NotFound);
+const SessionExpiredPage = fp.flow(withRouteData, withRoutingSpinner)(SessionExpired);
+const SignedInPage = fp.flow(withRouteData, withRoutingSpinner)(SignedIn);
+const SignInAgainPage = fp.flow(withRouteData, withRoutingSpinner)(SignInAgain);
+const SignInPage = fp.flow(withRouteData, withRoutingSpinner)(SignIn);
+const UserDisabledPage = fp.flow(withRouteData, withRoutingSpinner)(UserDisabled);
+
+interface RoutingProps {
+  onSignIn: () => void;
+  signIn: () => void;
+  subscribeToInactivitySignOut: () => void;
+  signOut: () => void;
+}
+
+const checkBrowserSupport = () => {
```

Comment:
app-routing doesn't seem like the right location for any of this stuff (browser, auth, override, bindClients, API override) - is there a reason it must go here? Comment or follow-up ticket?

This is all really just initial page load stuff